modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj

```diff
@@ -371,6 +371,29 @@
           ;; make sure that the fake exception was thrown, and thus the query execution was retried
           (is (true? @fake-execute-called)))))))
 
+(deftest not-retry-cancellation-exception-test
+  (mt/test-driver :bigquery-cloud-sdk
+    (let [fake-execute-called (atom false)
+          orig-fn        @#'bigquery/execute-bigquery]
+      (testing "Should not retry query on cancellation"
+        (with-redefs [bigquery/execute-bigquery (fn [^BigQuery client ^String sql parameters _ _]
+                                                  ;; We only want to simulate exception on the query that we're testing and not on possible db setup queries
+                                                  (if (and (re-find #"not-retry-cancellation-exception-test" sql) (not @fake-execute-called))
+                                                    (do (reset! fake-execute-called true)
+                                                        ;; Simulate a cancellation happening
+                                                        (throw (ex-info "Query cancelled" {::cancelled? true})))
+                                                    (orig-fn client sql parameters nil nil)))]
+          (try
+            (qp/process-query {:native {:query "SELECT CURRENT_TIMESTAMP() AS not-retry-cancellation-exception-test"} :database (mt/id)
```

Comment:
It's unusual to having dashes in names without quoting, I wonder if this works...

Reply:
Adjusted it since its indeed unusual